### ISSUE TITLE
View Available Hardware_Interface Handles

### ISSUE THREAD

**Issue by TomShkurti** [ros-controls/ros_control#346]
It would be helpful both for debugging and for familiarization with new robots (in cases where documentation is dubious or lacking) if there was a way to list all exposed hardware_interface handles and their types, similar to the controller_manager's `list` and `list-types` options.

**Comment by bmagyar** [comment]
That is indeed something I had in mind for a long time now.
We'd need to add support for this to `RobotHW` and `CombinedRobotHW`.

**Comment by littlejohnyang** [comment]
Yes I think this is definitely useful. I'm currently running into a problem that my own controller is complaining missing hardware interface (which I created myself). I really wanna see if it's actually available or not in order to debug what I have done wrong.

**Comment by bmagyar** [comment]
You can find this in one of the resourcemanager instances within the controller_manager, it should be easy to expose.
